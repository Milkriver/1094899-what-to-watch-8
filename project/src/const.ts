export enum AppRoute {
  Main = '/',
  SignIn = '/login',
  MyList = '/mylist',
  Film = '/films/:id',
  AddReview = '/films/:id/review',
  Player = '/player/:id'
}

export enum AuthorizationStatus {
  Auth = 'AUTH',
  NoAuth = 'NO_AUTH',
  Unknown = 'UNKNOWN',
}

export enum APIRoute {
  Movies = '/films',
  SingleMovie = '/films/:id',
  SameGenreMovies = '/films/:id/similar',
  PromoMovie = '/promo',
  FavouriteMovies = '/favorite',
  StatusFavouriteMovie = '/favorite/: film_id/: status',
  Reviews = '/comments/:id',
  Login = '/login',
  Logout = '/logout',
}
