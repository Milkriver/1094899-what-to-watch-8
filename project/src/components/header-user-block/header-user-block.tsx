import { connect, ConnectedProps } from 'react-redux';
import { Link } from 'react-router-dom';
import { AppRoute } from '../../const';
import { logoutAction } from '../../store/api-actions';
import { ThunkAppDispatch } from '../../types/actions';
import { IState } from '../../types/state';


const mapStateToProps = ({ authorizationStatus }: IState) => ({
  authorizationStatus,
});

const mapDispatchToProps = (dispatch: ThunkAppDispatch) => ({
  onSubmit() {
    return dispatch(logoutAction());
  },
});
const connector = connect(mapStateToProps, mapDispatchToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

function HeaderUserBlock(props: PropsFromRedux): JSX.Element {
  const { onSubmit } = props;
  const { authorizationStatus } = props;
  return (

    (authorizationStatus === 'AUTH') ?

      < ul className="user-block">
        <li className="user-block__item">
          <div className="user-block__avatar">
            <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
          </div>
        </li>
        <li className="user-block__item">
          <div onClick={onSubmit} className="user-block__link">Sign out</div>
        </li>
      </ul>
      :
      <div className="user-block">
        <Link to={AppRoute.SignIn} className="user-block__link">Sign in</Link>
      </div>


  );
}

export { HeaderUserBlock };
export default connector(HeaderUserBlock);
