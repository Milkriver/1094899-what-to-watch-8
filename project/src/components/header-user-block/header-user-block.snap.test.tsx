import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';
import { AuthorizationStatus } from '../../const';
import { HeaderUserBlock } from './header-user-block';

const mockAuthInfo = {
  id: 1,
  email: '',
  name: '',
  avatarUrl: '',
  token: '',
};
const mockFn = () => Promise.resolve();
describe('Component: HeaderUserBlock', () => {

  test('should render correctly', () => {
    const history = createMemoryHistory();
    const page = (
      <Router history={history}>
        <HeaderUserBlock authorizationStatus= {AuthorizationStatus.Auth} userData= {mockAuthInfo} onSubmit={mockFn}/>;
      </Router>
    );
    const { container } = render(page);
    expect(container).toMatchSnapshot();
  });
});
