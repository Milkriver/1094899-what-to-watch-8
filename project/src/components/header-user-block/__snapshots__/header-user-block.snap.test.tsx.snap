// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: HeaderUserBlock should render correctly 1`] = `
<div>
  <ul
    class="user-block"
  >
    <li
      class="user-block__item"
    >
      <div
        class="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="img/avatar.jpg"
          width="63"
        />
      </div>
    </li>
    <li
      class="user-block__item"
    >
      <div />
      <div
        class="user-block__link"
      >
        Sign out
      </div>
    </li>
  </ul>
  ;
</div>
`;
